torch/signal/windows/windows.py

```diff
@@ -296,3 +297,83 @@ def gaussian(
                        requires_grad=requires_grad)
 
     return torch.exp(-k ** 2)
+
+
+@_add_docstr(
+    r"""
+Computes the Kaiser window
+
+The Kaiser window is defined as follows:
+
+.. math::
+    out_i = I_0 \left( \beta \sqrt{1 - \left( {\frac{i - N/2}{N/2}} \right) ^2 } \right) / I_0( \beta )
+
+Where :math:`I_0` is the zeroth order modified Bessel function of the first kind (see :func:`torch.i0`) and
+``N = M - 1`` if :attr:`sym` is True and ``M`` if :attr:`sym` is False,
+where :attr:`M` is the window length.
+
+    """,
+    r"""
+
+{normalization}
+
+Args:
+    {M}
+
+Keyword args:
+    beta (float, optional): shape parameter for the window. Default: 12.0
+    {sym}
+    {dtype}
+    {layout}
+    {device}
+    {requires_grad}
+
+Examples::
+
+    >>> # Generate a symmetric gaussian window with a standard deviation of 1.0.
+    >>> torch.signal.windows.kaiser(5)
+    tensor([4.0065e-05, 2.1875e-03, 4.3937e-02, 3.2465e-01, 8.8250e-01, 8.8250e-01, 3.2465e-01, 4.3937e-02, 2.1875e-03, 4.0065e-05])
+
+    >>> # Generate a periodic gaussian window and standard deviation equal to 0.9.
+    >>> torch.signal.windows.kaiser(5,sym=False,std=0.9)
+    tensor([1.9858e-07, 5.1365e-05, 3.8659e-03, 8.4658e-02, 5.3941e-01, 1.0000e+00, 5.3941e-01, 8.4658e-02, 3.8659e-03, 5.1365e-05])
+""".format(
+        **window_common_args,
+    ),
+)
+def kaiser(
+        M: int,
```

Comment:
So this module is moving some of the positional SciPy arguments to be kwarg-only, but it's also providing defaults for them? In particular, `std` in `gaussian` above has a new default, and so does `beta` in `kaiser` here.

Reply:
IMO, if these defaults are reasonable (and a-priori they are), I'm happy with that, but I'll let @alvgaona assess.

Reply:
Not saying we shouldn't do it, just verifying my understanding

Reply:
Yes, we're making it to kwarg-only as well as giving it a default value. The SciPy implementations aren't consistent in this matter; if you look at `scipy.signal.windows.gaussian`, `std` is a required arg which is completely different from `scipy.signal.windows.exponential` for the parameter `tau`, so we're improving in that sense. 

In terms of kwarg-only, I believe that this kind of function is used with hardcoded values most of the time so to make the call a little bit more readable, enforcing the parameter name in the call sounds very reasonable.

For instance, 

```python
torch.signal.windows.gaussian(10, 2)
```

isn't as clear as

```python
torch.signal.windows.gaussian(10, std=2)
```

You could argue that the first argument is a required arg, but a convention is that the first parameter is always the size of the window, which is easy to remember.